### PR TITLE
Corrected f_beta computation

### LINKED ISSUE #4187: F beta with macro computation outputs nans for certain inputs
## Please reproduce using [the BoringModel and post here](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing)

https://colab.research.google.com/drive/1108mj1bKob0QpGlq2Q09L7ulmPDE49D1?usp=sharing

### To Reproduce

```bash
p = torch.Tensor([[0,0],[0,1]])
t = torch.Tensor([[0,1],[1,0]])
f_beta = Fbeta(num_classes=len(t[0]), average='macro', multilabel=True)
f_beta(p, t)

> tensor(nan)
```
### Expected behavior

tensor(0)

### Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```
 - PyTorch Lightning Version: 1.0.1
 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by abhinavg97**
Added METRIC_EPS in the denominator of computation to avoid nan values in f_beta score.

## What does this PR do?

Will give the correct f_beta score and prevent outputting nan values for multilabel classification with macro.

Fixes #4187 

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃 :+1:

### CODE REVIEW

**Review by Borda — CHANGES_REQUESTED** [review]
I think that this offset epsilon adding to all is not good, we shall check it there are zeros and then add it just there, well if know that some are rore just retunt the output...
@justusschock @SkafteNicki

**Comment by SkafteNicki** [comment]
We have this function (still used in functional) that took care of this:
https://github.com/PyTorchLightning/pytorch-lightning/blob/130de22fd75253330320473d8081fd60698c3d64/pytorch_lightning/metrics/functional/reduction.py#L40-L78
We should probably reuse it in the new class api

**Comment by SkafteNicki** [comment]
@abhinavg97 is it correct that this happens when both `recall` and `precision` is 0 for a particular class?
Could we add a test case for this?

**Comment by abhinavg97** [comment]
Yes, that's right. 

Here is an example:
```
p = torch.Tensor([[0,0],[0,1]])
t = torch.Tensor([[0,1],[1,0]])
```
Yeah I think we can add a test for this. Is it okay if I add the above case as a test?

**Comment by Borda** [comment]
can we add a test to verify that these changes fix the actual master...?

**Comment by SkafteNicki** [comment]
@abhinavg97 can you add the following testcase to `tests/metrics/classification/inputs` and afterwards add to the corresponding parametrization of the f-beta tests:
```
# Generate edge multilabel edge case, where nothing matches (scores are undefined)
__temp_preds = torch.randint(high=2, size=(NUM_BATCHES, BATCH_SIZE, NUM_CLASSES))
__temp_target = abs(__temp_preds - 1)

_multilabel_inputs_no_match = Input(
    preds=__temp_preds,
    target=__temp_target
)
```

**Comment by abhinavg97** [comment]
> @abhinavg97 can you add the following testcase to `tests/metrics/classification/inputs` and afterwards add to the corresponding parametrization of the f-beta tests:
> 
> ```
> # Generate edge multilabel edge case, where nothing matches (scores are undefined)
> __temp_preds = torch.randint(high=2, size=(NUM_BATCHES, BATCH_SIZE, NUM_CLASSES))
> __temp_target = abs(__temp_preds - 1)
> 
> _multilabel_inputs_no_match = Input(
>     preds=__temp_preds,
>     target=__temp_target
> )
> ```

Yes sure, will do. Thanks for the pointer to the input location :)

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Review by tchaton — APPROVED** [review]
Great PR ! Thanks !

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/metrics/classification/f_beta.py [hunk 1]
```diff
@@ -124,9 +125,11 @@ def compute(self):
             precision = self.true_positives.sum().float() / (self.predicted_positives.sum() + METRIC_EPS)
```
Comment (Borda) [inline pytorch_lightning/metrics/classification/f_beta.py:125]: do we really want to keep our metric systematically unprecise by adding some offset?

Comment (SkafteNicki) [reply pytorch_lightning/metrics/classification/f_beta.py:125]: we can move towards `class_reduce` function which explicit handles `nans`, when we unify the class based metrics and functional metrics.

Comment (abhinavg97) [reply pytorch_lightning/metrics/classification/f_beta.py:125]: yeah, makes sense. Just checked and the tests pass without METRIC_EPS. Pushing the update now.

**Review by Borda — APPROVED** [review]
